tests/apps/perf/pubsub_subscribe_http/subscribe.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+)
+
+func subscribeHandler(w http.ResponseWriter, r *http.Request) {
+	subscriptions := []subscription{}
+
+	subscribeType := os.Getenv("SUBSCRIBE_TYPE")
+	if subscribeType == "bulk" {
+		subscriptions = append(subscriptions, subscription{
+			PubsubName: pubSubName,
+			Topic:      topic + "-bulk",
+			Route:      route + "-bulk",
+			Metadata:   map[string]string{"bulkSubscribe": "true"},
+		})
+	} else {
+		subscriptions = append(subscriptions, subscription{
+			PubsubName: pubSubName,
+			Topic:      topic,
+			Route:      route,
+		})
+	}
+
+	log.Printf("Sending subscriptions: %v", subscriptions)
+
+	jsonBytes, err := json.Marshal(subscriptions)
+	if err != nil {
+		log.Fatalf("error marshalling subscriptions: %s", err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_, err = w.Write(jsonBytes)
+	if err != nil {
+		log.Fatalf("error writing response: %s", err)
+	}
+}
+
+func bulkMessageHandler(w http.ResponseWriter, r *http.Request) {
+	postBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Fatalf("error reading request body: %s", err)
+	}
+
+	var bulkSubscribeMessage bulkSubscribeMessage
+	err = json.Unmarshal(postBody, &bulkSubscribeMessage)
+	if err != nil {
+		log.Fatalf("error unmarshalling request body: %s", err)
+	}
+
+	log.Printf("received %d messages", len(bulkSubscribeMessage.Entries))
+	messagesCh <- len(bulkSubscribeMessage.Entries)
```

Comment:
should message uniqueness be considered as part of perf? 
As in measuring all the events have been received (p99) vs there being some duplicate events?
probably publish a json event with a unique ID field and have a map/set of IDs seen?

Reply:
> should message uniqueness be considered as part of perf? 
Isn't that a **correctness** test, which should be an e2e test scenario. WDYT? These checks will introduce further latency.

Reply:
But isn't perf supposed to measure complete response time only? And not partial which will happen on retries?

Reply:
TODO: Only count unique messages in subscriber application.